### PR TITLE
update event recorder to generated client

### PR DESCRIPTION
**Pull request by deads2k**
The event recorder (which should probably be removed, since no one actually uses its features) is the most widespread use of the manual client.  Remove that usage.

@sttts lots of changes, but basically all boilerplate.  Found this to be a pain when I started removing the manual client entirely.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32594)

### CODE REVIEW

File: pkg/client/record/event_test.go [hunk 1]
```diff
@@ -58,36 +58,36 @@ func (t *testEventSink) Update(e *api.Event) (*api.Event, error) {
 }
 
 // PatchEvent records the event for testing.
```
Comment (sttts) [inline pkg/client/record/event_test.go]: s/the event/a fake event/

File: pkg/controller/podautoscaler/horizontal.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ func newInformer(controller *HorizontalController, resyncPeriod time.Duration) (
 
 func NewHorizontalController(evtNamespacer unversionedcore.EventsGetter, scaleNamespacer unversionedextensions.ScalesGetter, hpaNamespacer unversionedautoscaling.HorizontalPodAutoscalersGetter, metricsClient metrics.MetricsClient, resyncPeriod time.Duration) *HorizontalController {
 	broadcaster := record.NewBroadcaster()
-	broadcaster.StartRecordingToSink(&unversionedcore.EventSinkImpl{Interface: evtNamespacer.Events("")})
+	broadcaster.StartRecordingToSink(evtNamespacer.Events(""))
```
Comment (sttts) [inline pkg/controller/podautoscaler/horizontal.go]: strange name: namespacer. Any reason why it's not called client here? (yes, not really part of the PR).

File: pkg/controller/scheduledjob/controller.go [hunk 3]
```diff
@@ -58,11 +59,11 @@ type ScheduledJobController struct {
 	recorder   record.EventRecorder
 }
 
-func NewScheduledJobController(kubeClient *client.Client) *ScheduledJobController {
+func NewScheduledJobController(kubeClient *client.Client, eventClient coreclient.EventsGetter) *ScheduledJobController {
 	eventBroadcaster := record.NewBroadcaster()
 	eventBroadcaster.StartLogging(glog.Infof)
 	// TODO: remove the wrapper when every clients have moved to use the clientset.
-	eventBroadcaster.StartRecordingToSink(kubeClient.Events(""))
+	eventBroadcaster.StartRecordingToSink(eventClient.Events(""))
```
Comment (sttts) [inline pkg/controller/scheduledjob/controller.go]: Is it intentional that you add a separate event client in some controller, but not all?

File: test/integration/scheduler/extender_test.go [hunk 4]
```diff
@@ -33,6 +33,7 @@ import (
 	"k8s.io/kubernetes/pkg/api/resource"
 	"k8s.io/kubernetes/pkg/api/testapi"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
```
Comment (sttts) [inline test/integration/scheduler/extender_test.go]: everywhere else this import is renamed to `clientset`

File: test/integration/scheduler/extender_test.go [hunk 5]
```diff
@@ -196,6 +197,7 @@ func TestSchedulerExtender(t *testing.T) {
 	defer framework.DeleteTestingNamespace(ns, s, t)
 
 	restClient := client.NewOrDie(&restclient.Config{Host: s.URL, ContentConfig: restclient.ContentConfig{GroupVersion: testapi.Default.GroupVersion()}})
+	clientset := internalclientset.NewForConfigOrDie(&restclient.Config{Host: s.URL, ContentConfig: restclient.ContentConfig{GroupVersion: testapi.Default.GroupVersion()}})
```
Comment (sttts) [inline test/integration/scheduler/extender_test.go]: elsewhere is mostly `kubeClient`

**Comment by sttts** [comment]
A number of consistency nits.

File: test/integration/scheduler/extender_test.go (hunk 4 above)
Comment (deads2k) [inline test/integration/scheduler/extender_test.go]: > everywhere else this import is renamed to clientset

Internal versus external is going to matter as we start trying to use the external clients we're vendoring in.  I'd like to start making the distinction.

File: pkg/controller/scheduledjob/controller.go (hunk 3 above)
Comment (deads2k) [inline pkg/controller/scheduledjob/controller.go]: > Is it intentional that you add a separate event client in some controller, but not all?

Yeah.  I added it to those that were still using the manual client and used the available interface elsewhere.

File: pkg/controller/podautoscaler/horizontal.go (hunk 2 above)
Comment (deads2k) [inline pkg/controller/podautoscaler/horizontal.go]: > strange name: namespacer. Any reason why it's not called client here? (yes, not really part of the PR).

Hold over name.

**Comment by deads2k** [comment]
Names updated.  I think the package makes a difference and `internalclientset` is more precise given our stated goal of moving to external clients.  I'd like to leave that import.

**Comment by sttts** [comment]
Even with the new review features, GH has no diff-of-diff support. Please don't squash before review is finished.

**Comment by deads2k** [comment]
I'm having some trouble getting e2e passing.  I may have to break this up and do it in passes :(
